vms/platformvm/service.go

```diff
@@ -1911,34 +1912,79 @@ func (service *Service) GetTx(_ *http.Request, args *api.GetTxArgs, response *ap
 // GetTxStatusArgs ...
 type GetTxStatusArgs struct {
 	TxID ids.ID `json:"txID"`
+	// If IncludeReason is false returns a response that looks like:
+	// {
+	// 	"jsonrpc": "2.0",
+	// 	"result": "Dropped",
+	// 	"id": 1
+	// }
+	// If IncludeReason is true returns a response that looks like this:
+	// {
+	// 	"jsonrpc": "2.0",
+	// 	"result": {
+	//     "status":"[Status]",
+	//     "reason":"[Reason tx was dropped, if applicable]"
+	//  },
+	// 	"id": 1
+	// }
+	// In the latter, "reason" is only present if the status is dropped
+	IncludeReason bool `json:"includeReason"`
+}
+
+// GetTxStatusResponse ...
+type GetTxStatusResponse struct {
+	includeReason bool
+	Status        `json:",omitempty"`
+	// Reason this tx was dropped.
+	// Only non-empty if Status is dropped
+	Reason string `json:"reason"`
+}
+
+func (r GetTxStatusResponse) MarshalJSON() ([]byte, error) {
+	if !r.includeReason {
+		return stdjson.Marshal(r.Status)
+	}
+	if r.Reason != "" {
+		return []byte(fmt.Sprintf("{\"status\": \"%s\", \"reason\": \"%s\"}", r.Status, r.Reason)), nil
+	}
+	return []byte(fmt.Sprintf("{\"status\": \"%s\"}", r.Status)), nil
```

Comment:
I think this should just return `\"%s\", r.Status` right?

Reply:
fixed